[tool.poetry]
name = "fmeval"
version = "1.2.2"
description = "Amazon Foundation Model Evaluations"
license = "Apache License 2.0"
authors = ["Amazon FMEval Team <amazon-fmeval-team@amazon.com>"]
packages = [
    { include = "fmeval", from = "src" },
]
readme = "README.md"
classifiers=[
    "Development Status :: 1 - Planning",
    "Intended Audience :: Developers",
    "Natural Language :: English",
    "Programming Language :: Python",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
]


[tool.poetry.dependencies]
python = "^3.10"
urllib3 = ">=2.3.0"
ray = "2.40.0"
semantic-version = "2.10.0"
pyarrow = "*"
pyfunctional = "1.5.0"
torch = ">=2.5.0"
matplotlib = "^3.10.0"
# https://discuss.ray.io/t/pandas-importerror-with-ray-data-dataset-show/13486
pandas = "2.2.3"
nltk = "^3.9.0"
markdown = "*"
IPython = "*"
evaluate = "^0.4.0"
rouge-score = "^0.1.2"
bert-score = "^0.3.13"
scikit-learn = "^1.6.0"
jiwer = "^3.0.5"
transformers = "^4.47.0"
sagemaker = "^2.237.1"
testbook = "^0.4.2"
ipykernel = "^6.29.5"
mypy-boto3-bedrock = "^1.35.75"
grpcio = "^1.68.1"
aiohttp = "^3.11.11"
tornado = "^6.4.2"

[tool.poetry.group.dev.dependencies]
fire = "*"
black = "24.10.0"
pre-commit = "^4.0.0"
pytest = "*"
pytest-pspec = "*"
flake8 = "*"
mypy = "*"
lxml = "*"
coverage = "*"
commitizen = "*"
conventional-pre-commit = "*"

[tool.poetry.group.docs]
optional = true

[tool.poetry.group.docs.dependencies]
pdoc = "^15.0.1"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
